server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java

```diff
@@ -0,0 +1,354 @@
+            return notExecutedResp(prev);
+         }
+      } else return notExistResp();
+   }
+
+   Response putIfAbsent() {
+      byte[] prev = cache.get(key);
+      if (prev == null) {
+         // Generate new version only if key not present
+         prev = cache.putIfAbsent(key, (byte[]) operationDecodeContext, buildMetadata());
+      }
+      if (prev == null)
+         return successResp(prev);
+      else
+         return notExecutedResp(prev);
+   }
+
+   Response put() {
+      // Get an optimised cache in case we can make the operation more efficient
+      byte[] prev = cache.put(key, (byte[]) operationDecodeContext, buildMetadata());
+      return successResp(prev);
+   }
+
+   EntryVersion generateVersion(ComponentRegistry registry, Cache<byte[], byte[]> cache) {
+      VersionGenerator cacheVersionGenerator = registry.getVersionGenerator();
+      if (cacheVersionGenerator == null) {
+         // It could be null, for example when not running in compatibility mode.
+         // The reason for that is that if no other component depends on the
+         // version generator, the factory does not get invoked.
+         NumericVersionGenerator newVersionGenerator = new NumericVersionGenerator()
+         .clustered(registry.getComponent(RpcManager.class) != null);
+         registry.registerComponent(newVersionGenerator, VersionGenerator.class);
+         return newVersionGenerator.generateNew();
+      } else {
+         return cacheVersionGenerator.generateNew();
+      }
+   }
+
+   Response remove() {
+      byte[] prev = cache.remove(key);
+      if (prev != null)
+         return successResp(prev);
+      else
+         return notExistResp();
+   }
+
+   Response removeIfUnmodified() {
+      CacheEntry<byte[], byte[]> entry = cache.getCacheEntry(key);
+      if (entry != null) {
+         byte[] prev = entry.getValue();
+         NumericVersion streamVersion = new NumericVersion(params.streamVersion);
+         if (entry.getMetadata().version().equals(streamVersion)) {
+            boolean removed = cache.remove(key, prev);
+            if (removed)
+               return successResp(prev);
+            else
+               return notExecutedResp(prev);
+         } else {
+            return notExecutedResp(prev);
+         }
+      } else {
+         return notExistResp();
+      }
+   }
+
+   Response clear() {
+      cache.clear();
+      return successResp(null);
+   }
+
+   Response successResp(byte[] prev){
+      return decoder.createSuccessResponse(header, prev);
+   }
+
+   Response notExecutedResp(byte[] prev) {
+      return decoder.createNotExecutedResponse(header, prev);
+   }
+
+   Response notExistResp() {
+      return decoder.createNotExistResponse(header);
+   }
+
+   Response createGetResponse(CacheEntry<byte[], byte[]> entry) {
+      return decoder.createGetResponse(header, entry);
+   }
+
+//   Object createMultiGetResponse(pairs: Map[Bytes, CacheEntry[Bytes, Bytes]]): AnyRef = null
+
```

Comment:
Yeah it was never used before either :P.  Just wanted to confirm I didn't forget this, heh.